### PR TITLE
CSS parsing optimizations

### PR DESCRIPTION
**Pull request by jdm**
These address some small inefficiencies that showed up while profiling the myspace talos test.

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17967)

### CODE REVIEW

File: components/style/properties/properties.mako.rs [hunk 1]
```diff
@@ -421,12 +421,11 @@ impl CSSWideKeyword {
...
-    fn parse<'i, 't>(_context: &ParserContext, input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i>> {
-        let ident = input.expect_ident()?.clone();
-        input.expect_exhausted()?;
-        CSSWideKeyword::from_ident(&ident)
-            .ok_or(SelectorParseError::UnexpectedIdent(ident).into())
+impl CSSWideKeyword {
+    fn parse(input: &mut Parser) -> Result<Self, ()> {
+        let ident = input.expect_ident().map_err(|_| ())?.clone();
+        input.expect_exhausted().map_err(|_| ())?;
+        CSSWideKeyword::from_ident(&ident).ok_or(())
```
Comment (emilio) [inline components/style/properties/properties.mako.rs]: Huh, why is this needed? I'd expect rustc to be able to figure this stuff out.

**Review by emilio — APPROVED** [review]

File: components/style/properties/properties.mako.rs (hunk 1 above)
Comment (jdm) [inline components/style/properties/properties.mako.rs]: Why is what needed, exactly?

Comment (emilio) [inline components/style/properties/properties.mako.rs]: Replacing the error by an smaller one... I'd expect this function to get inlined, and the error to disappear at all...

**Comment by wlach** [comment]
Platform microbenchmarks in gecko measured improvements from this change: https://treeherder.mozilla.org/perf.html#/alerts?id=8554

**Comment by bholley** [comment]
Nice!
